includes/class-wc-gateway-payfast.php

```diff
@@ -273,9 +272,12 @@ public function generate_payfast_form( $order_id ) {
 		}
 
 		$payfast_args_array = array();
+        $sign_strings = [];
 		foreach ( $this->data_to_send as $key => $value ) {
+            $sign_strings[] = esc_attr( $key ) . '=' . urlencode(str_replace('&amp;', '&', trim( $value )));
```

Comment:
A condition can be added here to ensure `$key !== 'source'` instead of completely removing it.

Reply:
Noted.  Will submit an updated patch.